workflow_scripts/test_models.py

```diff
@@ -6,81 +6,122 @@
+            for file in files:
+                if file.endswith(tar_ext_name) or file.endswith(onnx_ext_name):
+                    model_list.append(os.path.join(root, file))
+    return model_list
+
+
+def get_changed_models():
+    model_list = []
     cwd_path = Path.cwd()
     # git fetch first for git diff on GitHub Action
-    subprocess.run(['git', 'fetch', 'origin', 'main:main'],
+    subprocess.run(["git", "fetch", "origin", "main:main"],
                    cwd=cwd_path, stdout=subprocess.PIPE,
                    stderr=subprocess.PIPE)
     # obtain list of added or modified files in this PR
-    obtain_diff = subprocess.Popen(['git', 'diff', '--name-only', '--diff-filter=AM', 'origin/main', 'HEAD'],
+    obtain_diff = subprocess.Popen(["git", "diff", "--name-only", "--diff-filter=AM", "origin/main", "HEAD"],
                                    cwd=cwd_path, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
     stdoutput, _ = obtain_diff.communicate()
     diff_list = stdoutput.split()
 
     # identify list of changed ONNX models in ONXX Model Zoo
-    tar_ext_name = '.tar.gz'
-    onnx_ext_name = '.onnx'
     model_list = [str(model).replace("b'", "").replace("'", "")
                   for model in diff_list if onnx_ext_name in str(model) or tar_ext_name in str(model)]
+    return model_list
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Test settings")
+    # default all: test by both onnx and onnxruntime
+    # if target is specified, only test by the specified one
+    parser.add_argument("--target", required=False, default="all", type=str,
+                        help="Test the model by which (onnx/onnxruntime)?",
+                        choices=["onnx", "onnxruntime", "all"])
+    # use python workflow_scripts\test_models.py --create --all_models --keep to create broken test data by ORT
+    parser.add_argument("--create", required=False, default=False, action="store_true",
+                        help="Create new test data by ORT if it fails with existing test data")
+    parser.add_argument("--all_models", required=False, default=False, action="store_true",
+                        help="Test all ONNX Model Zoo models instead of only chnaged models")
+    parser.add_argument("--keep", required=False, default=False, action="store_true",
+                        help="Keep downloaded models after verification")
+    args = parser.parse_args()
+
+    model_list = get_all_models() if args.all_models else get_changed_models()
     # run lfs install before starting the tests
     test_utils.run_lfs_install()
 
-    print('\n=== Running ONNX Checker on added models ===\n')
-    # run checker on each model
+    print("\n=== Running test on ONNX models ===\n")
     failed_models = []
     for model_path in model_list:
-        model_name = model_path.split('/')[-1]
-        print('==============Testing {}=============='.format(model_name))
+        model_name = model_path.split("/")[-1]
```

Comment:
also enabling the 'black' formatting?

Reply:
Thanks for bringing this up. I was also thinking applying more Python lint/formatter in the CI and improving existing code. Will do this in another PR. At least for now, the added code in this PR pass flake8 except too long warnings.